### ISSUE TITLE
12factor app doesn't work (gke, digitalocean)

### ISSUE THREAD

**Issue by jimmykarily** [epinio/epinio#523]
First we had #521 . I tried to disable load balancing on gke (check the network settings when creating a cluster in the web ui) to see if that allows Epinio to work but after successfully pushing an application, the application route still gives a `502
Bad Gateway`. There are some errors that might be relevant in the traefik pod:

```
...
│ traefik time="2021-06-22T12:28:58Z" level=error msg="Cannot create service: subset not found" ingress=factor12 namespace=workspace providerName=kubernetes serviceName=factor12 servicePort=8080                                           │
...
│ linkerd-proxy [  1002.635758s]  WARN ThreadId(01) server{orig_dst=10.4.0.12:8080}: linkerd_app_core::errors: Failed to proxy request: error trying to connect: Connection refused (os error 111) client.addr=10.4.0.5:55472                │
```
After fixing #521 maybe no longer need to disable load balancing and things just work. We need to test this.

**Comment by jimmykarily** [comment]
It also needed #526 but even with both #526 and #521 in, the apps are still not accessible.

**Comment by jimmykarily** [comment]
It turns out it has something to do with the app. [12factor](https://github.com/heroku/12factor) doesn't work. The `assets/sample-app` in the Epinio repo and the [example-wordpress](https://github.com/epinio/example-wordpress) both work. We need to debug the 12factor app.

**Comment by manno** [comment]
The heroku sample app might not listen on IPv4 by default: https://github.com/epinio/epinio/issues/429#issuecomment-850487303

**Comment by agracey** [comment]
I think this was solved in slack but https://github.com/scf-samples/12factor does work.

**Comment by rohitsakala** [comment]
Tested on GKE using the two install command approach and it worked with scf-samples 12 factor app.

Tried installing with the single command install on GKE but failed. So, created an issue in `traefik` repo https://github.com/traefik/traefik/issues/8227. Waiting for their reply

**Comment by rohitsakala** [comment]
Created a PR to replace sample app with 12factor as requested. #541

**Comment by manno** [comment]
@rohitsakala why replace the existing phpinfo app with such a big test case? Wasn't the story about adding it to the repo?

**Comment by rohitsakala** [comment]
@manno Yes, I was confused about the story's acceptance criteria. Closed the PR

**Comment by rohitsakala** [comment]
The traefik issue https://github.com/traefik/traefik/issues/8227 was moved to https://community.traefik.io/t/cant-get-external-ip-when-i-use-loadbalanceripin-gke/11000. 

There is detailed thread there. 
TLDR- I have to use regional static IP address to install epinio in one single command in GKE using `loadBalancerIP`field in `traefik` service.

Also, created a new repo with 12factor app - https://github.com/epinio/example-12factor

Closing this issue now.
